comid/measurement_test.go

```diff
@@ -112,3 +121,216 @@ func TestMeasurement_NewUUIDMeasurement_bad_uuid(t *testing.T) {
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+		fmt.Printf("CBOR: %x\n", actual)
+	}
+}
+func TestMkey_MarshalCBOR_uint_notok(t *testing.T) {
+	tvs := []struct {
+		expected string
+	}{
+		{
+			expected: "unknown measurement key type: <nil>",
+		},
+	}
+
+	for _, tv := range tvs {
+		mkey := &Mkey{}
+		_, err := mkey.MarshalCBOR()
+
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestMkey_UnmarshalCBOR_uint_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     []byte
+		expected uint64
+	}{
+		{
+			mkey:     MustHexDecode(t, "00"),
+			expected: testMKeyUintMin,
+		},
+		{
+			mkey:     MustHexDecode(t, "1902BC"),
+			expected: TestMKey,
+		},
+		{
+			mkey:     MustHexDecode(t, "1BFFFFFFFFFFFFFFFF"),
+			expected: testMKeyUintMax,
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+
+		err := mKey.UnmarshalCBOR(tv.mkey)
+		assert.Nil(t, err)
+		actual, err := mKey.GetKeyUint()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+	}
+}
+
+func TestMkey_UnmarshalCBOR_uint_notok(t *testing.T) {
+	tvs := []struct {
+		mkey     []byte
+		expected string
+	}{
+		{
+			mkey:     []byte{0xAB, 0xCD},
+			expected: "unexpected EOF",
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+
+		err := mKey.UnmarshalCBOR(tv.mkey)
+
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestMkey_MarshalJSON_uint_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     uint64
+		expected []byte
+	}{
+		{
+			mkey:     testMKeyUintMin,
+			expected: []byte(`{"type":"uint","value":0}`),
+		},
+		{
+			mkey:     TestMKey,
+			expected: []byte(`{"type":"uint","value":700}`),
+		},
+		{
+			mkey:     testMKeyUintMax,
+			expected: []byte(`{"type":"uint","value":18446744073709551615}`),
+		},
+	}
+
+	for _, tv := range tvs {
+
+		meas := NewUintMeasurement(tv.mkey)
+		require.NotNil(t, meas)
+
+		actual, err := meas.Key.MarshalJSON()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+
+		fmt.Printf("JSON: %x\n", actual)
+	}
+}
+
+func TestMkey_MarshalJSON_uint_notok(t *testing.T) {
+	tvs := []struct {
+		expected string
+	}{
+		{
+			expected: "unknown type <nil> for mkey",
+		},
+	}
+
+	for _, tv := range tvs {
+
+		mkey := &Mkey{}
+
+		_, err := mkey.MarshalJSON()
+
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestMkey_UnmarshalJSON_uint_ok(t *testing.T) {
+	tvs := []struct {
+		input    []byte
+		expected uint64
+	}{
+		{
+			input:    []byte(`{"type":"uint","value":0}`),
+			expected: testMKeyUintMin,
+		},
+		{
+			input:    []byte(`{"type":"uint","value":700}`),
+			expected: TestMKey,
+		},
+		{
+			input:    []byte(`{"type":"uint","value":18446744073709551615}`),
+			expected: testMKeyUintMax,
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+
+		err := mKey.UnmarshalJSON(tv.input)
+		assert.Nil(t, err)
+		actual, err := mKey.GetKeyUint()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+	}
+}
+
+func TestMkey_UnmarshalJSON_uint_notok(t *testing.T) {
```

Comment:
I'd expect the test vectors here to be such that GetKeyUint fails rather than the failure happening on the generic mkey unmarshaler